File: backend/src/main/java/com/senac/ApiEduPlataformSenac/domain/entities/Professor.java
package com.senac.ApiEduPlataformSenac.domain.entities;

import com.senac.ApiEduPlataformSenac.domain.valueobjects.CPF;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Professor {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nome;

    @Embedded
    private CPF cpf;

    private String biografia;

    @OneToOne
    @JoinColumn(name = "usuario_id")
    private Usuario usuario;
}

File: frontend/app/types/professor.ts
export class Professor {
    constructor(
        public id: number | null,
        public nome: string,
        public cpf: string,
        public biografia: string
    ) { }
}

export interface ProfessorRequest {
    nome: string
    cpf: string
    biografia: string
}

export interface ProfessorResponse {
    id: number
    nome: string
    cpf: string
    biografia: string
}